packages/ai-jsx/src/lib/asr/vad.tsx

```diff
@@ -0,0 +1,117 @@
+'use client';
+import VADBuilder, { VAD, VADMode, VADEvent } from '@ozymandiasthegreat/vad';
+
+/**
+ * Base class for voice activity detectors.
+ */
+export abstract class VoiceActivityDetectorBase {
```

Comment:
If every field is `abstract`, I think you just want an `interface` here instead.

Reply:
good point. note that the callbacks are not abstract though.